In-memory caching

GRA0007 opened the pull request:
Closes #17 

This PR uses `keyv` to provide in-memory caching based on a key and an optional `ttl` parameter. It can be used in the following way:

```js
export default {
  'express/js/route/:withParams': {
    handler: yourRouteHandlerFunction,
    key: request => ['A key to cache with', request.params.withParams], // optional, any type, will cache result based on key
    ttl: 3600000, // time to live, optional, in milliseconds
  }
}
```

Future plans would include storing the cache to a file or database, which can be set up with the keyv library ([see docs](https://www.npmjs.com/package/keyv))

giraugh reviewed (changes requested):
This is really cool! I have a little bit of feedback however.
I think some of the internal naming might be a bit confusing. I reckon we make these changes (happy to do these changes if you like and approve)

`RouteHandler` -> `RouteHandlerFn`
`RouteObject` -> `RouteHandlerObject`
`RouteValue` -> `RouteHandler`

`getRouteObject` -> `routeHandlerAsObject`
`resolveHandler` -> `applyHandler` (or maybe `useHandler`? not sure on this one)

I also think that maybe the `key` and `ttl` parameters could be `cacheKey` and `cacheTTL` but not sure on that. Whats your thoughts?

Would it also be usefull to have a way to enable/disable caching for all routes? Like a '--caching false' CLI configuration option?

But otherwise super hyped for this and I think `keyv` is a great call.

GRA0007 commented:
I've updated those type and function names, good call they were confusing before. I've also added a cmd option to disable caching, `--no-cache`. I reckon `key` and `ttl` are fine, they both relate directly to the result of the handler.

giraugh reviewed (approved):
Cool! Thanks for making those changes :) Is looking good :+1:
